libc/src/__support/CPP/type_traits/is_destructible.h

```diff
@@ -16,12 +16,11 @@
 #include "src/__support/CPP/type_traits/true_type.h"
 #include "src/__support/CPP/type_traits/type_identity.h"
 #include "src/__support/macros/attributes.h"
-#include "src/__support/macros/config.h"
```

Comment:
This PR allows us to remove the dependency on `libc.src.__support.macros.config` from `type_traits` in libc/src/__support/CPP/CMakeLists.txt, and `:__support_macros_config` from `__support_cpp_type_traits` in utils/bazel/llvm-project-overlay/libc/BUILD.bazel.

Reply:
Done.